opal/mca/accelerator/rocm/accelerator_rocm_module.c

```diff
@@ -573,6 +576,23 @@ static int mca_accelerator_rocm_open_ipc_handle(int dev_id, opal_accelerator_ipc
     return OPAL_SUCCESS;
 }
 
+static int mca_accelerator_rocm_compare_ipc_handles(uint8_t handle_1[IPC_MAX_HANDLE_SIZE],
+                                                    uint8_t handle_2[IPC_MAX_HANDLE_SIZE])
+{
+    /*
+     * The HIP IPC handles consists of multiple elements.
+     * We will only use the ROCr IPC handle (32 bytes, starting at pos 0)
+     * and the process ID for comparison.
+     * We definitily need to exclude the offset component in the comparison.
+     */
+    static const int rocr_ipc_handle_size = 32;
+    static const int pos = rocr_ipc_handle_size + 2*sizeof(size_t);
+    int *pid_1 = (int *)&handle_1[pos];
+    int *pid_2 = (int *)&handle_2[pos];
+
+    return (memcmp(handle_1, handle_2, 32) + (*pid_1 == *pid_2 ? 0 : 1));
```

Comment:
Ah this might not work because `-1 + 1 = 0`

Reply:
:-( you are unfortunately right again. Thanks for catching it!

Reply:
A more readable way can be
```
...
    if (*pid_1 != *pid_2) return *pid_1 > *pid_2;
    return memcmp(handle_1, handle_2, 32);
...
```

Reply:
that is actually a good idea, I made just one minor change, thanks!